src/components/task-step/step-mixin.cjsx

```diff
@@ -23,8 +26,20 @@ module.exports =
   render: ->
     footer = @renderFooterButtons?() or @renderGenericFooter()
     {pinned} = @props
-
-    <CardBody className='task-step' footer={footer} pinned={pinned}>
+    <CardBody className={@state.classNames.join(' ')} footer={footer} pinned={pinned}>
       {@renderBody()}
       {@renderGroup?()}
     </CardBody>
+
+  componentDidMount:  ->
+    @setBannerClass()
+
+  setBannerClass: ->
+    has_banner = _.contains(@state.classNames, "with-ui-banner")
+    if not has_banner and this.getDOMNode().querySelector(":first-child[data-label]")
+      # The defer's needed to break out of the rendering loop so the store has time to
+      # mark itself as loaded
+      # TODO: investigate why
+      _.delay =>
+        @setState classNames: @state.classNames.concat ["with-ui-banner"] if @isMounted()
+      , 20
```

Comment:
I feel like this is "wrong" on many levels :grin:

The main thing that's complicating the code is that the "card-body" needs to have extra top margin if a banner is going to be displayed, otherwise it'll be rendered under the top navbar.

But the code has no way of knowing what the content looks like until it renders it.  A previous attempt of mine tried running a regex on it in the Reading component, but that was pretty brittle and felt even more wrong.

The other things that's weird is that their are two loaded events fired, one for the tasks endpoint, and then for the steps.  This causes the components to mount/unmount in rapid succession.  That's the root cause for needing the `.delay` and  `isMounted` checks.  I haven't investigated much further at this point - am hopeful that someone can point out a better way and I can remove this method altogether.